ide/static/js/signup.js

```diff
@@ -0,0 +1,298 @@
+import React from "react";
+import DjangoCSRFToken from "django-react-csrftoken";
+import $ from "jquery";
+
+class Signup extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      error_username: 0,
+      error_email: 0,
+      error_password1: 0,
+      error_password2: 0,
+      correct: "#8d9ab0",
+      wrong: "#eb4d4b",
+      label: "#3f4a5c"
+    };
+    this.submitForm = this.submitForm.bind(this);
+    this.validateEmail = this.validateEmail.bind(this);
+    this.validateUsername = this.validateUsername.bind(this);
+    this.validatePassword1 = this.validatePassword1.bind(this);
+    this.validatePassword2 = this.validatePassword2.bind(this);
+  }
+  validateUsername() {
+    let username = document.getElementById("username-value").value;
+    let form_error_message = document.getElementById("form-error-message");
+    if (username.length > 150) {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.wrong;
+      let username_label = document.getElementById("username-label");
+      username_label.style.fontSize = "0.7em";
+      username_label.style.top = "-1em";
+      username_label.style.left = "2.4em";
+      form_error_message.innerHTML =
+        "The username can't be more than 150 characters.";
+      form_error_message.style.opacity = 1;
+      username_label.style.color = this.state.wrong;
+      this.setState({ error_username: 1 });
+    } else if (username.length == "") {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.wrong;
+      form_error_message.innerHTML = "The username can't be empty.";
+      form_error_message.style.opacity = 1;
+      let username_label = document.getElementById("username-label");
+      username_label.style.color = this.state.wrong;
+      this.setState({ error_username: 1 });
+    } else {
+      document.getElementById(
+        "username-value"
+      ).style.borderColor = this.state.correct;
+      document.getElementById("username-label").style.color = this.state.label;
+      form_error_message.style.opacity = 0;
+      this.setState({ error_username: 0 });
+    }
+  }
+  validateEmail() {
+    let email = document.getElementById("email-value").value;
+    let form_error_message = document.getElementById("form-error-message");
+    if (!email.includes("@")) {
+      document.getElementById(
+        "email-value"
+      ).style.borderColor = this.state.wrong;
+      let email_label = document.getElementById("email-label");
+      if (email != "") {
+        email_label.style.fontSize = "0.7em";
```

Comment:
For style changes based on actions can't you create separate css class json in the component and change the style class of component ?

Reply:
Sure, I'll keep this in mind when coding the new design.